### DISCUSSION TITLE
SPF / DKIM checks for hosted domain is bypassed by DMS? (Bitcoin extortion scam)

### DISCUSSION THREAD

**Discussion by kub3let** [docker-mailserver/docker-mailserver#3687] [Q&A] [answered]
I'm currently getting a ton of "Bitcoin Extortion Scam" from some asian botnet, some of them go through the sieve filter.

What makes me wonder about them is they originate from the same domain, e.g. they fake the from header and make it seem like you sent it yourself like if your email is admin@example.com they are from admin@example.com to admin@example.com.

I wonder how this can be since my domain has everything in place, spf, dmarc, dkim, dane, etc. these emails should be blocked right away since spf etc. fails. In addition the botnet is also in several blacklists.

I would like for docker-mailserver to properly block those faked emails, I could just add custom sieve rules or decrease the sieve spam score but that would not fix the problem at the root.

server log:
```
Dec  7 09:04:57 mail postfix/smtpd[324804]: connect from unknown[xx.xx.xx.xx]
Dec  7 09:04:59 mail policyd-spf[327291]: prepend Received-SPF: None (mailfrom) identity=mailfrom; client-ip=xx.xx.xx.xx; helo=[0.0.0.0]; envelope-from=admin@example.com; receiver=<UNKNOWN> 
Dec  7 09:04:59 mail postfix/smtpd[324804]: BADFAXXXXXXX: client=unknown[xx.xx.xx.xx]
Dec  7 09:05:06 mail postfix/cleanup[327294]: BADFAXXXXXXX: message-id=<32XXXX.321XXXX@24331.com>
Dec  7 09:05:06 mail opendkim[2388]: BADFAXXXXXXX: [xx.xx.xx.xx] [xx.xx.xx.xx] not internal
Dec  7 09:05:06 mail opendkim[2388]: BADFAXXXXXXX: not authenticated
Dec  7 09:05:06 mail opendkim[2388]: BADFAXXXXXXX: no signature data
Dec  7 09:05:06 mail opendmarc[2397]: BADFAXXXXXXX: example.com none
Dec  7 09:05:06 mail postfix/qmgr[2535]: BADFAXXXXXXX: from=<admin@example.com>, size=2766, nrcpt=1 (queue active)
Dec  7 09:05:06 mail postfix/smtpd[324804]: disconnect from unknown[xx.xx.xx.xx] ehlo=1 mail=1 rcpt=1 data=1 commands=4
Dec  7 09:05:10 mail postfix/smtpd-amavis/smtpd[327313]: connect from localhost[127.0.0.1]
Dec  7 09:05:10 mail postfix/smtpd-amavis/smtpd[327313]: 83E12XXXXXX: client=localhost[127.0.0.1]
Dec  7 09:05:10 mail postfix/cleanup[327294]: 83E12XXXXXX: message-id=<32XXXX.321XXXX@24331.com>
Dec  7 09:05:10 mail postfix/smtpd-amavis/smtpd[327313]: disconnect from localhost[127.0.0.1] ehlo=1 mail=1 rcpt=1 data=1 quit=1 commands=5
Dec  7 09:05:10 mail postfix/qmgr[2535]: 83E12XXXXXX: from=<admin@example.com>, size=3611, nrcpt=1 (queue active)
Dec  7 09:05:10 mail amavis[318788]: (318788-02) Passed CLEAN {RelayedInbound}, [xx.xx.xx.xx]:35042 [175.157.165.61] <admin@example.com> -> <admin@example.com>, Queue-ID: BADFAXXXXXXX, Message-ID: <32XXXX.321XXXX@24331.com>, mail_id: XXXXXXXXXX, Hits: 3.511, size: 2823, queued_as: 83E12XXXXXX, 4127 ms
Dec  7 09:05:10 mail postfix/smtp-amavis/smtp[327304]: BADFAXXXXXXX: to=<admin@example.com>, relay=127.0.0.1[127.0.0.1]:10024, delay=12, delays=8/0.01/0/4.1, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as 83E12XXXXXX)
Dec  7 09:05:10 mail postfix/qmgr[2535]: BADFAXXXXXXX: removed
Dec  7 09:05:10 mail dovecot: lmtp(327315): Connect from local
Dec  7 09:05:10 mail dovecot: lmtp(admin@example.com)<327315><4MZPIXXXXX/gQA9XXXXX>: sieve: msgid=<32XXXX.321XXXX@24331.com>: stored mail into mailbox 'INBOX'
Dec  7 09:05:10 mail dovecot: lmtp(327315): Disconnect from local: Logged out (state=READY)
Dec  7 09:05:10 mail postfix/lmtp[327314]: 83E12XXXXXX: to=<admin@example.com>, relay=mail.example.com[/var/run/dovecot/lmtp], delay=0.09, delays=0.02/0.02/0.01/0.
04, dsn=2.0.0, status=sent (250 2.0.0 <admin@example.com> 4MZPIXXXXX/gQA9XXXXX Saved)
Dec  7 09:05:10 mail postfix/qmgr[2535]: 83E12XXXXXX: removed
```

email:
```
Return-Path: <admin@example.com>
Delivered-To: admin@example.com
Received: from mail.example.com
	by mail.example.com with LMTP
	id xxxxxxxxx/xxxxxxxx
	(envelope-from <admin@example.com>)
	for <admin@example.com>; Thu, 07 Dec 2023 09:05:10 +0000
Received: from localhost (localhost [127.0.0.1])
	by mail.example.com (Postfix) with ESMTP id xxxxxxxxxxxx
	for <admin@example.com>; Thu,  7 Dec 2023 09:05:10 +0000 (UTC)
X-Spam-Flag: NO
X-Spam-Score: 3.511
X-Spam-Level: ***
X-Spam-Status: No, score=3.511 tagged_above=2 required=6.31
	tests=[BITCOIN_EXTORT_01=0.001, BITCOIN_SPAM_07=1.772,
	BITCOIN_TOEQFM=2.139, FSL_HELO_NON_FQDN_1=0.001, GB_BITCOIN_CP=0.344,
	HELO_NO_DOMAIN=0.001, PDS_BTC_ID=0.499, PYZOR_CHECK=1.985,
	RCVD_IN_DNSWL_HI=-5, RCVD_IN_ZEN_BLOCKED_OPENDNS=0.001,
	RDNS_NONE=1.274, SPF_NONE=0.001, SUBJ_ALL_CAPS=0.5,
	T_SCC_BODY_TEXT_LINE=-0.01, URIBL_BLOCKED=0.001,
	URIBL_DBL_BLOCKED_OPENDNS=0.001, URIBL_ZEN_BLOCKED_OPENDNS=0.001]
	autolearn=no autolearn_force=no
Received-SPF: None (mailfrom) identity=mailfrom; client-ip=xx.xx.xx.xx; helo=[0.0.0.0]; envelope-from=admin@example.com; receiver=<UNKNOWN> 
Authentication-Results: mail.example.com; dmarc=none (p=none dis=none) header.from=example.com
Received: from [0.0.0.0] (unknown [xx.xx.xx.xx])
	by mail.example.com (Postfix) with ESMTP id xxxxxxx
	for <admin@example.com>; Thu,  7 Dec 2023 09:04:58 +0000 (UTC)
Received: from xxxx ([175.157.165.61]) by 24331.com with MailEnable ESMTP; Thu, 7 Dec 2023 17:05:05 +0800
Received: (qmail 32122 invoked by uid 321); 7 Dec 2023 17:05:03 +0800
From: admin@example.com
To: admin@example.com
Subject: YOUR DEVICE AND EMAIL HAS BEEN COMPROMISED CHECK THIS MESSAGE NOW!
Date: Thu, 7 Dec 2023 17:05:05 +0800
Message-ID: <xxxxx.xxxxx@24331.com>
Mime-Version: 1.0
Content-type: text/plain;

Hello there!

Unfortunately, there are some bad news for you.

Some time ago your device was infected with my private trojan, R.A.T (Remote Administration Tool), if you want to find out more about it simply use Google.

...
```

docker-mailserver setup:
```
docker run -d \
--name mail \
--hostname mail \
--domainname example.com \
--net dmz --ip xx.xx.xx.xx  \
--ip6 "xxxxxxxxxxxxxxxxxxx" \
-v /etc/localtime:/etc/localtime:ro \
-v /srv/mail/maildata:/var/mail \
-v /srv/mail/mailstate:/var/mail-state \
-v /srv/mail/config/:/tmp/docker-mailserver \
-v /srv/proxy/certs/:/etc/letsencrypt:ro \
-e SA_KILL=1000.00 \
-e SPAMASSASSIN_SPAM_TO_INBOX=1 \
-e MOVE_SPAM_TO_JUNK=1 \
-e ENABLE_SPAMASSASSIN=1 \
-e ENABLE_CLAMAV=1 \
-e ENABLE_FAIL2BAN=0 \
-e ENABLE_POSTGREY=0 \
-e ONE_DIR=1 \
-e DMS_DEBUG=0 \
-e SSL_TYPE=letsencrypt \
-e TLS_LEVEL=intermediate \
-e SPOOF_PROTECTION=1 \
-e POSTFIX_MESSAGE_SIZE_LIMIT=33554432 \
-e POSTFIX_MAILBOX_SIZE_LIMIT=5368709120 \
-e ENABLE_UPDATE_CHECK=0 \
--cpus="2" --memory="4096m" --memory-swap="4096m" \
--restart unless-stopped \
mailserver/docker-mailserver
```

**Comment by polarathene** [comment]
Just to confirm, was the IPv4 address in the logs by any chance from the Docker network gateway? (eg: `172.16.0.1`)

By default I don't think any service should be allowing any connection from within a subnet, but that is something to be mindful of (_eg: A reverse proxy routing public traffic to DMS_).

Also note that you should really only be using `--hostname`, not `--domainname` (NIS Domain).

---

Our [v13 release notes](https://github.com/docker-mailserver/docker-mailserver/releases/tag/v13.0.0) have a less helpful change description regarding `SA_KILL` collapsed under the "Merged Pull Requests" section, but I made sure to better communicate this in the `CHANGELOG.md`:

https://github.com/docker-mailserver/docker-mailserver/blob/908d38047ce4be6dd1e8ca5021d0bce86625a3c3/CHANGELOG.md?plain=1#L93

This could be why your spam is getting through as `SA_KILL` has no effect due to `SPAMASSASSIN_SPAM_TO_INBOX=1`? (_but your intent with `MOVE_SPAM_TO_JUNK=1` seems to be to [send that to Junk folder anyway](https://github.com/docker-mailserver/docker-mailserver/blob/908d38047ce4be6dd1e8ca5021d0bce86625a3c3/target/scripts/startup/setup.d/security/misc.sh#L267-L269)_)

The warning should appear in your logs when you follow our [troubleshooting advice](https://docker-mailserver.github.io/docker-mailserver/latest/config/debugging/#steps-for-debugging-dms) to increase the `LOG_LEVEL` for more insights. I see you have `DMS_DEBUG=0` which was deprecated a while back and [replaced with `LOG_LEVEL` in v11](https://github.com/docker-mailserver/docker-mailserver/blob/master/CHANGELOG.md#v1100).

You've not indicated what version of DMS you're running, filling out a proper bug report would probably be best.

---

> ```
> Dec  7 09:04:59 mail policyd-spf[327291]: prepend Received-SPF: None (mailfrom) identity=mailfrom; client-ip=xx.xx.xx.xx; helo=[0.0.0.0]; envelope-from=admin@example.com; receiver=<UNKNOWN> 
> Dec  7 09:04:59 mail postfix/smtpd[324804]: BADFAXXXXXXX: client=unknown[xx.xx.xx.xx]
> Dec  7 09:05:06 mail postfix/cleanup[327294]: BADFAXXXXXXX: message-id=<32XXXX.321XXXX@24331.com>
> Dec  7 09:05:06 mail opendkim[2388]: BADFAXXXXXXX: [xx.xx.xx.xx] [xx.xx.xx.xx] not internal
> Dec  7 09:05:06 mail opendkim[2388]: BADFAXXXXXXX: not authenticated
> Dec  7 09:05:06 mail opendkim[2388]: BADFAXXXXXXX: no signature data
> Dec  7 09:05:06 mail opendmarc[2397]: BADFAXXXXXXX: example.com none
> ```

This and anti-spam features aren't really my area of expertise in the project, so @casperklein or @georglauterbach may have a better idea why these are failing to cause Postfix to reject the mail before Amavis + SA analyze it for spam detection.

It would be good to know if the rspamd integration works for you instead as it's intended to replace all 3 services (SPF / DKIM / DMARC) in the above logs. Perhaps there's something wrong with the configuration there, assuming the mail is not bypassing checks (_eg: If the source address is within the same subnet_).

**Reply by kub3let** [reply]
@polarathene thanks, I highly appreciate your reply.


> Just to confirm, was the IPv4 address in the logs by any chance from the Docker network gateway? (eg: `172.16.0.1`)

no, also I use macvlan so the container has a full network stack so I don't think that should even be possible.


> By default I don't think any service should be allowing any connection from within a subnet, but that is something to be mindful of (_eg: A reverse proxy routing public traffic to DMS_).

I actually moved this week to a  temporary setup where the traffic is masqueraded and routed from a secondary firewall, but it's still a whole different subnet.

The originating ip is 10.40.0.10/24 while the ip of docker-mailserver is 192.168.3.120/24 but given that the log prints `not internal` I doubt that should matter here ?


> Also note that you should really only be using `--hostname`, not `--domainname` (NIS Domain).
> The warning should appear in your logs when you follow our [troubleshooting advice](https://docker-mailserver.github.io/docker-mailserver/latest/config/debugging/#steps-for-debugging-dms) to increase the `LOG_LEVEL` for more insights. I see you have `DMS_DEBUG=0` which was deprecated a while back and [replaced with `LOG_LEVEL` in v11](https://github.com/docker-mailserver/docker-mailserver/blob/master/CHANGELOG.md#v1100).

Thanks I will update that.

> This could be why your spam is getting through as `SA_KILL` has no effect due to `SPAMASSASSIN_SPAM_TO_INBOX=1`? (_but your intent with `MOVE_SPAM_TO_JUNK=1` seems to be to [send that to Junk folder anyway](https://github.com/docker-mailserver/docker-mailserver/blob/908d38047ce4be6dd1e8ca5021d0bce86625a3c3/target/scripts/startup/setup.d/security/misc.sh#L267-L269)_)

Junk/Spam detection overall works and moves spam into the Junk folder, but for some reason in about 5% of these spam emails the spam score is too low and they are sent to Inbox, although only the BTC address seems to be different.

> You've not indicated what version of DMS you're running, filling out a proper bug report would probably be best.

mailserver/docker-mailserver      latest         906ae61e1177   8 days ago      622MB

---

I have setup a sieve filter for now which filters all scam mails by subject.

I also did this spoof test https://emailspooftest.com/ and the only thing failed is the from header check which is exactly the same from the spam I get, all other (8/9) mails got rejected.

Description: BEC fake insider protection
Failed :system did not protect against BEC & reply-chain, fake insider attacks
This vulnerability allows attackers to impersonate ANYONE FROM YOUR COMPANY. This is a highly attacked vulnerability.

**Reply by polarathene** [reply]
> ```
> Received-SPF: None (mailfrom) identity=mailfrom; client-ip=xx.xx.xx.xx; helo=[0.0.0.0]; envelope-from=admin@example.com; receiver=<UNKNOWN> 
> Authentication-Results: mail.example.com; dmarc=none (p=none dis=none) header.from=example.com
> Received: from [0.0.0.0] (unknown [xx.xx.xx.xx])
> 	by mail.example.com (Postfix) with ESMTP id xxxxxxx
> 	for <admin@example.com>; Thu,  7 Dec 2023 09:04:58 +0000 (UTC)
> Received: from xxxx ([175.157.165.61]) by 24331.com with MailEnable ESMTP; Thu, 7 Dec 2023 17:05:05 +0800
> ```

It might be worth trying [`POSTFIX_REJECT_UNKNOWN_CLIENT_HOSTNAME=1`](https://docker-mailserver.github.io/docker-mailserver/v13.0/config/environment/#postfix_reject_unknown_client_hostname)?:

https://github.com/docker-mailserver/docker-mailserver/blob/d3b4e94d0681b98ea6a7a955051c397cf19c0b88/target/scripts/startup/setup.d/postfix.sh#L67-L72

[Postfix docs](https://www.postfix.org/postconf.5.html#reject_unknown_client_hostname) (_I've slightly modified it in the quote_):

> Reject the request when any of these conditions are true:
> 1) the client IP address->name mapping fails
> 2) the name->address mapping fails
> 3) the name->address mapping does not match the client IP address.
> 
> This is a stronger restriction than the [`reject_unknown_reverse_client_hostname`](https://www.postfix.org/postconf.5.html#reject_unknown_reverse_client_hostname) feature, which triggers only under condition 1) above.

`reject_unknown_reverse_client_hostname` is probably sufficient in this case? The `helo` in the SPF check shouldn't be `0.0.0.0` though should it? Is this from the spoofer or DMS Postfix? Or is it the result of your network setup to the DMS container and FQDN?

Also potentially relevant, if your network has multiple interfaces I wonder if that comes into play? [There's a PR open currently](https://github.com/docker-mailserver/docker-mailserver/pull/3465) where Amavis needed to be configured to bind to a specific interface for the IP to match DNS records with outbound mail.

You're more experienced with networking than I am, but it's possible that your network differs from our own assumptions (happy path) which may be a source of issues.

> I also did this spoof test https://emailspooftest.com/ and the only thing failed is the from header check which is exactly the same from the spam I get, all other (8/9) mails got rejected.

I don't run DMS myself, but I assume your failing spoof test from https://emailspooftest.com/ is not affecting other users deployment of DMS 🤔 At least I think we'd have received complaints of that by now if that were the case.

---

> The originating ip is `10.40.0.10/24`

Isn't that in the private range? The actual sender would be from a public range (_which looks like it was `175.157.165.61`?_)

So is the problem more to do with this information was lost due to some proxy in your network between the spammer and DMS? That is a common issue that we have had reports of problems:
- Docker routing the public port from all public interfaces to internal private network port, changing the source address to that internal gateway due to `userland-proxy: true` Docker daemon config.
- HAProxy / Traefik routing TCP traffic without preserving the source address (_similar with other container runtimes like Podman or Rootless IIRC, our docs touch on that_).

That would lead to issues like Gmail failing to verify legit mail sent from DMS IIRC, or DoS attacks by having Fail2Ban trigger on that IP masking all external traffic coming in through it, since it was not aware of the original source address, preventing legit users from login.

> but given that the log prints not internal I doubt that should matter here ?

Not my area of expertise, I'd need to setup local instances to try reproduce 😓 

As long as Postfix has the correct information to perform the `postfix_reject_unknown_client_hostname` lookup, it should reject the mail. We did have this enabled by default at one point, but it was found to be too aggressive for some users while others found it worked very well.

**Reply by polarathene** [reply, marked as answer]
Would be good to know if that suggested ENV would have avoided the issue.

At least we've identified it is related to the suspected cause of the network environment and an intermediary between DMS and the client sending spam rewriting the client source address 👍 

The default should be `PERMIT_DOCKER=none` and this does not appear to have been changed from the provided config. [I don't see anything there](https://github.com/docker-mailserver/docker-mailserver/blob/d3b4e94d0681b98ea6a7a955051c397cf19c0b88/target/scripts/startup/setup.d/networking.sh) that would imply DMS is an open relay by default to local networks for that configuration. Rspamd was a concern as it's not handled by `PERMIT_DOCKER=none`, but that feature / service was not enabled.

Ideally if the issue is due to our config defaults we need to identify where so that can be fixed. Otherwise this concern is related to those in the bullet points of my prior comment and we could better document that in the troubleshooting/debugging docs page.

**Comment by georglauterbach** [comment]
This is what I was pinged for: "Have you tried Rspamd?" 😆

On a serious note: maybe using Rspamd helps here? It seems you are rather pleased with the current setup (Amavis/SA/etc.) but the 5% of cases annoys you; I am not confident that Rspamd improves the situation, but you could give it a shot.

What DNS resolver are you using? To be more precise: how does DMS _exactly_ resolve DNS queries?

**Reply by kub3let** [reply]
> What DNS resolver are you using? To be more precise: how does DMS _exactly_ resolve DNS queries?

Opnsense as forwarder using cloudflare via DOT but I do have split dns so there is a override for the local network which resolves *.example.com/example.com to the firewall address 192.168.3.1 which is then port forwarded to the mail server 192.168.3.120.

So I guess this could be a problem... If docker-mailserver resolves mail.example.com it looks like it is in the same subnet.

Still I don't see why this would be enabled by default, seems like a security flaw IMHO.

**Reply by georglauterbach** [reply]
1. Why is `.example.com` resolved to a local address (that presumably belongs to a firewall)? This is, IMO, the actual security flaw.
2. IIRC accesses to `*.example.com` are resolved to `192.168.3.1` and there is a port-forwarding so that you actually access the mail server (DMS?) at `192.168.3.120`? So DMS, doing a request to `*.example.com`, is accessing itself?
3. Having local networks on a whitelist is actually common practice (e.g., https://github.com/docker-mailserver/docker-mailserver/blob/77917f5cc64b444db2a76def92161186db66670b/target/rspamd/local.d/options.inc#L3). I have not worked with Amavis/SA on DMS for a long time - @casperklein can probably tell us more about whitelistes local subnets.

**Reply by polarathene** [reply]
> This is what I was pinged for: "Have you tried Rspamd?" 😆

Ha, sorry about that, it was more in regards to the DKIM / DMARC / SPF checks not rejecting the mail which from what I understood it should have done before processing through Amavis for a spam score (_which is a separate concern_).

@georglauterbach are you able to confirm if you also get the same failing spoof test (header `FROM`) at https://emailspooftest.com/

That'd probably be a good way of identifying if it's a configuration issue with DMS or configuration.

Regarding the DNS issue, the spammer would need to match the same assigned local IP for that to validate? That doesn't seem to be the case from what we can see (_`175.157.165.61`? Or is it the `0.0.0.0` which would match the `helo` from SPF?_)

---

Regarding the subnets, I agree with @kub3let that these should not be by default (_at least the private ranges within `10.x.y.z` / `172.x.y.z` / `192.168.x.y`_). There is an open TODO issue regarding the current [`PERMIT_DOCKER` implementation](https://github.com/docker-mailserver/docker-mailserver/issues/3478) and concerns there.

You'll notice that it will adjust what subnets are permitted for Postfix, OpenDKIM and OpenDMARC, but you skipped integrating Rspamd into that.
- Ideally `PERMIT_DOCKER` feature would properly control what subnets are allowed to bypass security checks for Rspamd too.
- The default is `none` which should otherwise be effective, but @kub3let has clarified they're using macvlan for their network which may not match some assumptions such as `NETWORK_INTERFACE` ENV that is tied to the `PERMIT_DOCKER` feature.

**Reply by georglauterbach** [reply]
Interesting: with Rspamd, I get E5 and E7 delivered (DKIM and SPF failed, but mail got delivered) - this is coherent with my DMS setup and default DMS setups should behave the same. #3323 is there to possibly change this.

This is not what OP is seeing, though. IMO, the issue of the OP is here:

> ```
> Received: from [0.0.0.0] (unknown [xx.xx.xx.xx])
> by mail.example.com (Postfix) with ESMTP id xxxxxxx
> for <admin@example.com>; Thu,  7 Dec 2023 09:04:58 +0000 (UTC)
> ```

This should not happen.

> `Received: from xxxx ([175.157.165.61]) by 24331.com with MailEnable ESMTP; ...`

Why the `xxxx`? I'd like to know what's written there because this is the initial sender. Clearing something there seems pointless to me.

---

What I need; this may be picky, but it lets maintainers assess the situation best:

1. A log of the mail being received where you anonymize the IP addresses but instead of using xx.xx.xx.xx you use technically valid IPs and tell us exactly who belongs to which IP. Please do not anonymize IPs that do not belong to yourself or local IPs!!
2. Please do the same for the e-mail contents (most importantly the headers, duh) :)

**Comment by georglauterbach** [comment]
@polarathene I have started investigating the matter. `PERMIT_DOCKER` is indeed fiddling around a lot, but its name is also outdated. I will propose a new approach with a new ENV.

Simultaneously, I have uncovered a bit about our current Rspamd configuration. This may help in pushing #3323.

**Comment by kub3let** [comment]
Ok so realized I'm an Idiot for having setup the masquerading, SPF is bypassed this way since everything looks to come from a local subnet, I have fixed the routing and everything looks good now, sorry I've wasted your time.

This still leaves a security vulnerably, since DMS allows everything from private networks by default which allows anyone (malicious user, malware, etc.) to impersonate anyone and send emails on their free will without authentication. DMS should never act as open relay, IP's should be explicitly whitelisted for it.

**Reply by georglauterbach** [reply]
You are correct. I have proposes a new way of dealing with local networks, see https://github.com/docker-mailserver/docker-mailserver/issues/3478#issuecomment-1847993688. This should eliminate the problem (or make it harder to configure DMS in such a way).

**Reply by polarathene** [reply]
> DMS allows everything from private networks by default which allows anyone (malicious user, malware, etc.) to impersonate anyone and send emails on their free will without authentication. DMS should never act as open relay

I don't think that's entirely the case? At least for Postfix, OpenDKIM and OpenDMARC they should be restricted by default (`PERMIT_DOCKER=none`). It would be good to identify where in the configuration this is not the case so we can release a bugfix for it.

Other than that, the `0.0.0.0` source address for a `HELO` looked odd, that's basically any interface AFAIK?

@georglauterbach are you aware of where in the config we've allowed DMS to be an open relay for local networks? @kub3let did not have Rspamd enabled, so it wouldn't have been there.

**Comment by georglauterbach** [comment]
**NOTE**: I strongly advise against using <https://emailspooftest.com/> because I am seeing the same fishing e-mails from `xxxxx.xxxxx@24331.com> now 😆 They have a score of 18.9 though, so they are outright rejected.
